### PR TITLE
Code bugs

### PR DESCRIPTION
**Pull request by KoenDG**
This very small PR is related to issue #24226 it is small exactly to keep it readable.

These are what appear to be actual bugs. In order:

1. Arrays printed without Arrays.toString(). Which results in the actual String being something like `[B@2f4a...` instead of the actual contents of the array.

2. Code that is not used at all. There doesn't appear to be any point in the code of these files that actually make use of the constructs being set up. In any way. So I'm assuming they were once used and not cleaned up completely when code was moved around.

3. I'm assuming the `logger.equals(msg)` bit it a typo. Judging by its surrounding methods, it should be `logger.error(msg)`.

===========

1. Contributor agreement: Has been signed

2. Contributor guidelines: I have read them.

3. Gradle check: build successful.

4. PR is against master

5. The PR is not OS specific.

6. I'm not part of a class.

### CODE REVIEW

File: modules/transport-netty4/src/main/java/org/elasticsearch/transport/netty4/Netty4InternalESLogger.java [hunk 1]
```diff
@@ -161,7 +161,7 @@ public boolean isErrorEnabled() {
 
     @Override
     public void error(String msg) {
-        logger.equals(msg);
+        logger.error(msg);
```
Comment (jasontedor) [inline modules/transport-netty4/src/main/java/org/elasticsearch/transport/netty4/Netty4InternalESLogger.java]: This is horrific, thank you for catching this, I need to take this one out of this PR right now and bring it all the way back to the 5.4 branch which I'm not likely to do with the remainder of these changes. Are you okay with me doing that?

**Review by jasontedor — APPROVED** [review]
LGTM, but I need to steal one of the changes right now!

File: modules/transport-netty4/src/main/java/org/elasticsearch/transport/netty4/Netty4InternalESLogger.java (hunk 1 above)
Comment (jasontedor) [inline modules/transport-netty4/src/main/java/org/elasticsearch/transport/netty4/Netty4InternalESLogger.java]: Also I committed this atrocity. 😢

Comment (jasontedor) [inline modules/transport-netty4/src/main/java/org/elasticsearch/transport/netty4/Netty4InternalESLogger.java]: I pushed 1b660c51275972e01520becafafc82110c8e50a0, can you merge master in to your branch and then I can pull these rest of these in?

Comment (nik9000) [inline modules/transport-netty4/src/main/java/org/elasticsearch/transport/netty4/Netty4InternalESLogger.java]: I've done worse.

**Comment by KoenDG** [comment]
@jasontedor master merged into branch and updated.

**Comment by jasontedor** [comment]
ok to test

**Comment by KoenDG** [comment]
@jasontedor I was going over some more results, if you're still here, can you verify something?

DecayFunctionParser.java line 107 and 109. Isn't that token being moved twice?

**Comment by jasontedor** [comment]
The `DecayFunctionParser` code is fine, it parses a field name, and then moves to either a `{` or the value of a `multi_value_mode` if that was the field name.

The `DocumentParser#parseObject` code needs improvement, that method always returns null as written and the return value is indeed never used.

> EDIT2: Sorry for using a PR like a quickchat.

Yes, it's probably best that we do not do this. I think [forum](https://discuss.elastic.co) topics are best.

**Comment by jasontedor** [comment]
Can you break this PR into three separate changes?
 - incorrect invocation of `Object#toString` on arrays
 - removing dead code in SnapshotsService
 - removing dead code in ScriptedMetricAggregationBuilder

Sorry, I know this will take more of your time and of our time but this is how we work and we need to maintain that here too.

**Comment by KoenDG** [comment]
Not a problem. It'll be a few hours before I'm at my home computer to do it though.

**Comment by KoenDG** [comment]
The 3 PRs have been made, that should be it. I'll close this one as any further discussion can happen in the respective PR.

**Pull request closed without merging**